pkgs/tools/admin/yandexcloud-cli/default.nix

```diff
@@ -0,0 +1,104 @@
+{ lib
+, stdenv
+, fetchurl
+, makeBinaryWrapper
+, installShellFiles
+, testers
+  # update script
+, writeShellScript
+, curl
+, common-updater-scripts
+}:
+let
+  version = "0.109.0";
+  hashes = {
+    "aarch64-darwin" = "sha256-Fyt41VyjWuaRlF4ulwSo2yTX9jlBh/+syv53UxrC+jE=";
+    "aarch64-linux" = "sha256-Dy3jxSxTBbuHJEKcchVlayIlUANkPMxuYcksO+18Te0=";
+    "i686-linux" = "sha256-yG2ZSdZ0c+zf3cAn4Kkomlck5EjS8I0MHcJndInW4Ik=";
+    "x86_64-darwin" = "sha256-iRB1+aYsXptR4ksIa0vtWIucx5aiqGy676fBDuvN/e4=";
+    "x86_64-linux" = "sha256-2Vul/LrcdGncwf571Px06AwBKAt/qmlIUvbWC6TtWkI=";
+  };
+
+  storageURL = "https://storage.yandexcloud.net/yandexcloud-yc/release";
+
+  fetchSource = system: hash:
+    let
+      throwSystem = throw "Unsupported system: ${system}";
+      platform = lib.systems.elaborate system;
```

Comment:
I should say this function looks too mysterious to me  and I never have seen it before.
(And it appears only 8 times in Nixpkgs.)

I will think on it.

Reply:
I’ve updated `fetchSource` to use simple `lib.splitString`.